physical/cassandra.go

```diff
@@ -0,0 +1,186 @@
+package physical
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	log "github.com/mgutz/logxi/v1"
+
+	"github.com/armon/go-metrics"
+	"github.com/gocql/gocql"
+)
+
+// CassandraBackend is a physical backend that stores data in Cassandra.
+type CassandraBackend struct {
+	sess  *gocql.Session
+	table string
+
+	logger log.Logger
+}
+
+// newCassandraBackend constructs a Cassandra backend using a pre-existing
+// keyspace and table.
+func newCassandraBackend(conf map[string]string, logger log.Logger) (Backend, error) {
+	splitArray := func(v string) []string {
+		return strings.FieldsFunc(v, func(r rune) bool {
+			return r == ','
+		})
+	}
+
+	var (
+		hosts       = splitArray(conf["hosts"])
+		keyspace    = conf["keyspace"]
+		table       = conf["table"]
+		consistency = gocql.LocalQuorum
+	)
```

Comment:
In fact, there's a really really good argument to make for factoring out code from the cassandra secret backend that parses parameters and sets up a client into a package in `helper/` :-D